### DISCUSSION TITLE
rsimple crashes on startup

### DISCUSSION THREAD

**Discussion by bclay1297** [pelicanmapping/rocky#45] [Q&A / Support]
I was able to build rocky with vspkg with Glens help.  The resulting files were put in the G:\build directory.
I set the environment variables as instructed
   set ROCKY_FILE_PATH=G:\build\vcpkg_installed\x64-windows\share
   set ROCKY_DEFAULT_FONT=C:/windows/fonts/arialbd.ttf
   set VSG_FILE_PATH=G:\build\vcpkg_installed\x64-windows\share

When I run rsimple I get the following messages
[vsg info] texel_margin = 12
[vsg info] quad_margin = 6
[rocky warning] Trouble: Rocky may not be able to find its shaders. Consider setting one of the environment variables VSG_FILE_PATH or ROCKY_FILE_PATH.
[rocky warning] TerrainNode initialize failed: Terrain shaders are missing or corrupt. Did you set ROCKY_FILE_PATH to point at the rocky share folder?

I am not sure that it is a related issue but if I cd into G:\rocky>cd G:\build\vcpkg_installed\x64-windows\bin
then run vsgqtviewer.exe or vsgqtwindows.exe nothing happens it just returns to the command line.  No errors show or anything
I am running on Windows 10 using Visual Studio 2022

Bruce

**Comment by gwaldron** [comment]
Did you build the `INSTALL` target?
Are the shaders in the install location under `share`?

**Comment by bclay1297** [comment]
Glenn:
  I just ran the bootstrap-vcpsg.bat and it created the build directory
where I tried to run. When I open the rocky visual studio solution I do not
see an install.

Bruce

**Reply by gwaldron** [reply]
Did you build the project? The "INSTALL" target is under "CMakePredefinedTargets" in Visual Studio.

Or, you can build from the command line with
```
cmake --build build --target install --config RelWithDebInfo
```

**Comment by bclay1297** [comment]
Glen:
  I did not initially see the install project as it is under
CmakePredefinedTargets.  After your suggestion I looked around and found
and ran INSTALL  It did not appear to help.

based on the website instructions I opened a command file and ran the
following env direction.
    set ROCKY_FILE_PATH=G:\build\vcpkg_installed\x64-windows/share
    set ROCKY_DEFAULT_FONT=C:/windows/fonts/arialbd.ttf
    set PROJ_DATA=G:\build\vcpkg_installed\x64-windows/share/proj.

This finally got rid of the error on proj,db which I had copied into the
bin directory where I am running from.

I looked in the share directory and did not see anything marked as a
shader.  I found a shader folder back in the initial download folder and
copied that to the share folder. . Now instead of crashing I get a window
with an orange ball in it.  Is that what I should see?
I get the following message in the console window

G:\build\vcpkg_installed\x64-windows\debug\bin>rsimple
[vsg info] texel_margin = 12
[vsg info] quad_margin = 6
[vsg info] Context::waitForCompletion()  0000021217015428 fence->wait()
failed with error. VkResult = -4

I ran rdemo and a globe appeared along with a menu on the left side.  After
a while though the app froze

The console window showed the following:
[vsg info] texel_margin = 12
[vsg info] quad_margin = 6
[rocky info] OGR feature source
https://readymap.org/readymap/filemanager/download/public/countries.geojson
opened OK
[rocky info] OGR feature source
https://readymap.org/readymap/filemanager/download/public/countries.geojson
opened OK
[rocky warning] [TMS] Layer "ReadyMap 90m Elevation" : SSL connection failed
[rocky warning] Problem getting data from "ReadyMap 90m Elevation" : SSL
connection failed

The window does not appear to be getting any updates.  If this is a Qt
based app I wonder if QApplication::ProcessEvents is still getting called.

It looks like we are gaining.

Thanks again for your help

Bruce
